DEPRECATION.md

```diff
@@ -0,0 +1,72 @@
+# Deprecation
+
+This page provides an overview of features that are deprecated in Cube.js.
+Changes in packaging, and supported (Linux) distributions are not included. To
+learn about end of support for Linux distributions, refer to the
+[changelog](CHANGELOG.md).
+
+## Feature Deprecation Policy
+
+As changes are made to Cube.js, there may be times when existing features need
+to be removed or replaced with newer features. Before an existing feature is
+removed it is marked as "deprecated" within the documentation and remains in
+Cube.js for at least one stable release unless specified explicitly otherwise.
+After that time it may be removed.
+
+Users are expected to take note of the list of deprecated features each release
+and plan their migration away from those features, and (if applicable) towards
+the replacement features as soon as possible.
+
+## Deprecated Features
+
+The table below provides an overview of the current status of deprecated
+features:
+
+- **Deprecated**: the feature is marked "deprecated" and should no longer be
+  used. The feature may be removed, disabled, or change behavior in a future
+  release. The _"Deprecated"_ column contains the release in which the feature
+  was marked deprecated, whereas the _"Remove"_ column contains a tentative
+  release in which the feature is to be removed. If no release is included in
+  the _"Remove"_ column, the release is yet to be decided on.
+- **Removed**: the feature was removed, disabled, or hidden. Refer to the linked
```

Comment:
I wonder if it would sound more developer-friendly if we reverse the logic here: talk about *keeping the feature until a certain date* instead of *feature removal at a certain date*. In this fashion, we might want to change all *remove-based* terms to "keep-until" variants. @hassankhan, WDYT?

Reply:
@igorlukanin I'm not sure I agree 😅

In my opinion, this is a deprecation document (as opposed to a support document), so we should be clear in communicating what our policies are in this regard. Of course, happy to change it if anyone has a different opinion 😀